x-pack/plugins/enterprise_search/public/applications/app_search/utils/results/index.ts

```diff
@@ -5,70 +5,70 @@
  * 2.0.
  */
 
-/* This method flattens fields in documents returned from the ent-search backend.
- * If a field in the document contains "raw" key, it's already flat.
- * If it doesn't, we want to pull properties from it, and move them to the top level.
- * This field is already flat:
- * 'country', { raw: 'United States' }
- * This field is not flat:
- * 'address', {
- *     country: { raw: 'United States' },
- *     city: { raw: 'Los Angeles' }
- * }
- * It will be transformed into:
- * [
- *   ['address.country', { raw: 'United States' }],
- *   ['address.city', { raw: 'Los Angeles' }]
- * ]
- */
-export const flattenDocumentField = (
-  fieldName: string,
-  fieldValue: object
-): Array<[string, object]> => {
-  const flattened: Array<[string, object]> = [];
+function isFieldValueWrapper(object: object): boolean {
```

Comment:
Question: Is it possible to use not use object here, but Record<string, unknown>?

Reply:
Also while you have this utility function possibly we can use it for narrowing with proper types instead of using "object"